easy-ui-react/src/Modal/useIntersectionDetection.ts

```diff
@@ -0,0 +1,36 @@
+import { RefObject, useEffect, useState } from "react";
+
+/**
+ * Detects when an element is intersecting with a scroll root.
+ *
+ * @param targetRef The target element to observe
+ * @param scrollRef The scroll root to check against
+ * @returns when element is intersecting
+ */
+export function useIntersectionDetection(
+  targetRef: RefObject<HTMLDivElement>,
+  scrollRef: RefObject<HTMLDivElement>,
+) {
+  const [isStuck, setIsStuck] = useState(false);
+
+  useEffect(() => {
+    if (!targetRef.current || !scrollRef.current) {
+      return;
+    }
+
+    const observer = new IntersectionObserver(
+      ([entry]) => {
+        setIsStuck(!entry.isIntersecting);
+      },
+      { root: scrollRef.current },
+    );
```

Comment:
**question**: hmmm im probably missing something obvious, but what is this bit doing?

`{ root: scrollRef.current }`

Reply:
it's passing in a custom scroll root to the intersection observer so that the intersection can be based off of a non-body scroll container, in the case of our modals the scroll root is the body of the Modal. does that make sense? i can maybe update the naming to reflect that better

Reply:
ahh! i see, yes that makes sense. `scrollRef` seems quite appropriate then, im good with the naming